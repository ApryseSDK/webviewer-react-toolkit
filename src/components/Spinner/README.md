Shows non-specific progress. Best used if there is an unknown amount of time
before the intended action completes.
