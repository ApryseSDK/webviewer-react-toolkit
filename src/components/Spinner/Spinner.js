'use strict';
Object.defineProperty(exports, '__esModule', { value: true });
exports.Spinner = void 0;
var tslib_1 = require('tslib');
var classnames_1 = tslib_1.__importDefault(require('classnames'));
var react_1 = tslib_1.__importDefault(require('react'));
var Spinner = function(_a) {
  var _b = _a.spinnerSize,
    spinnerSize = _b === void 0 ? 'default' : _b,
    className = _a.className;
  var spinnerClass = (0, classnames_1.default)(
    'ui__base ui__spinner',
    'ui__spinner--size-'.concat(spinnerSize),
    className,
  );
  return react_1.default.createElement(
    'div',
    { className: spinnerClass },
    react_1.default.createElement('div', {
      className: 'ui__spinner__animated',
    }),
  );
};
exports.Spinner = Spinner;
//# sourceMappingURL=Spinner.js.map
