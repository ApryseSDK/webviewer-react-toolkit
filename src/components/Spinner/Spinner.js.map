{"version":3,"file":"Spinner.js","sourceRoot":"","sources":["Spinner.tsx"],"names":[],"mappings":";;;;AAAA,kEAAoC;AACpC,wDAA0B;AAcnB,IAAM,OAAO,GAAG,UAAC,EAAoD;QAAlD,mBAAuB,EAAvB,WAAW,mBAAG,SAAS,KAAA,EAAE,SAAS,eAAA;IAC1D,IAAM,YAAY,GAAG,IAAA,oBAAU,EAAC,sBAAsB,EAAE,4BAAqB,WAAW,CAAE,EAAE,SAAS,CAAC,CAAC;IAEvG,OAAO,CACL,uCAAK,SAAS,EAAE,YAAY;QAC1B,uCAAK,SAAS,EAAC,uBAAuB,GAAG,CACrC,CACP,CAAC;AACJ,CAAC,CAAC;AARW,QAAA,OAAO,WAQlB"}