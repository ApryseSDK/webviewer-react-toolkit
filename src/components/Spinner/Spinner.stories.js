'use strict';
Object.defineProperty(exports, '__esModule', { value: true });
exports.Basic = void 0;
var tslib_1 = require('tslib');
var addon_knobs_1 = require('@storybook/addon-knobs');
var react_1 = tslib_1.__importDefault(require('react'));
var Spinner_1 = require('../Spinner');
var README_md_1 = tslib_1.__importDefault(require('./README.md'));
exports.default = {
  title: 'Components/Spinner',
  component: Spinner_1.Spinner,
  parameters: { readme: README_md_1.default },
};
var Basic = function() {
  return react_1.default.createElement(Spinner_1.Spinner, {
    spinnerSize: (0, addon_knobs_1.select)(
      'spinnerSize',
      ['tiny', 'small', 'default', 'large'],
      'default',
    ),
  });
};
exports.Basic = Basic;
//# sourceMappingURL=Spinner.stories.js.map
