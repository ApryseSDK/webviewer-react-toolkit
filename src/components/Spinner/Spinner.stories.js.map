{"version":3,"file":"Spinner.stories.js","sourceRoot":"","sources":["Spinner.stories.tsx"],"names":[],"mappings":";;;;AAAA,sDAAgD;AAChD,wDAA0B;AAC1B,sCAAqC;AACrC,kEAAiC;AAEjC,kBAAe,EAAE,KAAK,EAAE,oBAAoB,EAAE,SAAS,EAAE,iBAAO,EAAE,UAAU,EAAE,EAAE,MAAM,qBAAA,EAAE,EAAE,CAAC;AAEpF,IAAM,KAAK,GAAG,cAAM,OAAA,CACzB,8BAAC,iBAAO,IAAC,WAAW,EAAE,IAAA,oBAAM,EAAC,aAAa,EAAE,CAAC,MAAM,EAAE,OAAO,EAAE,SAAS,EAAE,OAAO,CAAC,EAAE,SAAS,CAAC,GAAI,CAClG,EAF0B,CAE1B,CAAC;AAFW,QAAA,KAAK,SAEhB"}