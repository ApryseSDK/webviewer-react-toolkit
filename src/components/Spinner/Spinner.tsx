import classnames from 'classnames';
import React, { FC } from 'react';

export enum SpinnerStyle {
  Default = 'default',
  OnGray = 'on-gray',
}

export enum SpinnerSize {
  Tiny = 'tiny',
  Small = 'small',
  Default = 'default',
  Large = 'large',
}

export interface SpinnerProps {
  /**
   * Set the visual appearance of the spinner.
   * @default SpinnerStyle.Default
   */
  spinnerStyle?: SpinnerStyle;
  /**
   * Set the size of the spinner.
   * @default SpinnerSize.Default
   */
  spinnerSize?: SpinnerSize;
  /**
   * Classname for the container div.
   */
  className?: string;
}

export const Spinner: FC<SpinnerProps> = ({
  spinnerStyle = SpinnerStyle.Default,
  spinnerSize = SpinnerSize.Default,
  className,
}) => {
  const spinnerClass = classnames(
    'ui__base ui__spinner',
    `ui__spinner--style-${spinnerStyle}`,
    `ui__spinner--size-${spinnerSize}`,
    className,
  );

  return (
    <div className={spinnerClass}>
      <div className="ui__spinner__animated" />
    </div>
  );
};
