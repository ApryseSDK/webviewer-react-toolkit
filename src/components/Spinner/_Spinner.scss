.ui__spinner {
  display: inline-block;

  &__animated {
    width: 100%;
    height: 100%;
    animation: spin 1.2s cubic-bezier(0.4, 0.2, 0.6, 0.8) infinite;
    border-radius: 50%;
    border: solid;
    border-color: $color-blue-gray-5 $color-blue-gray-1 $color-blue-gray-1;
  }

  /* --- Spinner sizes. --- */

  &--size {
    &-tiny {
      width: 12px;
      height: 12px;

      .ui__spinner__animated {
        border-width: 1px;
      }
    }

    &-small {
      width: 16px;
      height: 16px;

      .ui__spinner__animated {
        border-width: 2px;
      }
    }

    &-default {
      width: 24px;
      height: 24px;

      .ui__spinner__animated {
        border-width: 2px;
      }
    }

    &-large {
      width: 32px;
      height: 32px;

      .ui__spinner__animated {
        border-width: 2px;
      }
    }
  }
}
