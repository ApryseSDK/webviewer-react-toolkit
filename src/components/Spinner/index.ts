export { Spinner as default, SpinnerProps, SpinnerSize, SpinnerStyle } from './Spinner';
