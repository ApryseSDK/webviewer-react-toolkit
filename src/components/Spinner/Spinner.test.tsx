import { shallow } from 'enzyme';
import React from 'react';
import { Spinner } from '../Spinner';

describe('Spinner component', () => {
  it('renders its contents', () => {
    const spinner = shallow(<Spinner />);
    expect(spinner.find('.ui__spinner')).toHaveLength(1);
  });

  it('snapshot renders default spinner', () => {
    const spinner = shallow(<Spinner />);
    expect(spinner).toMatchSnapshot();
  });
});
