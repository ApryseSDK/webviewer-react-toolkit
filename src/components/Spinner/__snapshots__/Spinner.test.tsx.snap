// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner component snapshot renders default spinner 1`] = `
<div
  className="ui__base ui__spinner"
>
  <div
    className="ui__spinner__animated"
  />
</div>
`;
